recipes/implot/all/conanfile.py

```diff
@@ -24,41 +27,45 @@ class ImplotConan(ConanFile):
         "fPIC": True
     }
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        copy(self, "CMakeLists.txt", self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def requirements(self):
-        if tools.Version(self.version) >= "0.13":
-            self.requires("imgui/1.87")
+        if Version(self.version) >= "0.14":
+            self.requires("imgui/1.89.4", transitive_headers=True)
         else:
-            self.requires("imgui/1.86")
+            # imgui 1.89 renamed ImGuiKeyModFlags_* to  ImGuiModFlags_*
+            self.requires("imgui/1.88", transitive_headers=True)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.configure()
-        return cmake
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["IMPLOT_SRC_DIR"] = self.source_folder.replace("\\", "/")
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, os.pardir))
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
 
     def package_info(self):
```

Comment:
```suggestion
    def package_info(self):
        self.cpp_info.libs = ["implot"]
        self.cpp_info.requires = ["imgui::imgui"]
```
I thought that if the package had requirements they had to be specified in cpp_info ?

Probably it would be good to be explicit.

https://github.com/epezent/implot/blob/master/implot.h#L48

Reply:
In my understanding the requirements are set by self.requires. And since the dependency on imgui is a public one `transitive_headers` has to be set. See https://github.com/hobbeshunter/conan-center-index/blob/004a7f5c2eb62f76c47815f4c848c1263b355d87/recipes/implot/all/conanfile.py#L43

If I'm not mistaken `self.cpp_info.requires` will already be populated. I guess you only have to be specific if your package has components. Then conan can't guess: https://docs.conan.io/2.0/reference/conanfile/methods/package_info.html#components

But I'm still getting used to conan 2. So I might be wrong.

Reply:
I think you're totally right. I'm so used to the conan CLI requiring them on cpp_info.components that I didn't realize that for single-lib pkgs this is optional.

Reply:
single component == automatic :) you understand it pretty on point